x-pack/test/reporting/README.md

```diff
@@ -1,153 +0,0 @@
-## The Reporting Tests
```

Comment:
This README has a lot of information, but I think the only useful part is about how to update the baseline snapshots. The previous steps for doing so were a bit complicated - I've updated the snapshot in the past when I switched the tests to use an archived dashboard, and it was way easier than this README talks about. The simplified instructions are moved out to `x-pack/test/functional/apps/dashboard/reporting/README.md`

As of 8.0, Reporting no longer supports backwards-compatibility URLs, so all that info was obsolete.